### PR TITLE
MPP-2002 measure tracker removal feature usage

### PR DESCRIPTION
**Pull request by say-yawn**
# New feature description
Adding metrics on backend to measure usage of tracker removal feature by creating a `pre_save` signal and updating the tracker list with latest changes.

# Checklist

- [ ] l10n changes have been submitted to the l10n repository, if any.
- [ ] All acceptance criteria are met.
- [ ] I've added or updated relevant docs in the docs/ directory.
- [ ] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: emails/signals.py [hunk 1]
```diff
@@ -27,5 +33,32 @@ def check_premium_for_block_list_emails(sender, instance, **kwargs):
...
+        return
+    curr_profile = Profile.objects.get(id=instance.id)
+
+    # measure tracker removal usage
+    changed_tracker_removal_setting = (
+        instance.remove_level_one_email_trackers
+        != curr_profile.remove_level_one_email_trackers
+    )
+    if changed_tracker_removal_setting:
+        print(changed_tracker_removal_setting)
```
Comment (jwhitlock) [inline emails/signals.py:49]: Please remove this `print`

File: emails/signals.py [hunk 2]
```diff
@@ -27,5 +33,32 @@ def check_premium_for_block_list_emails(sender, instance, **kwargs):
...
+        if instance.remove_level_one_email_trackers:
+            incr_if_enabled("tracker_removal_enabled")
+        if not instance.remove_level_one_email_trackers:
+            incr_if_enabled("tracker_removal_disabled")
+        info_logger.info(
+            "tracker_removal_feature",
+            extra={
+                "enabled": instance.remove_level_one_email_trackers,
+                "hashed_uid": sha256(instance.fxa.uid.encode("utf-8")).hexdigest(),
```
Comment (jwhitlock) [inline emails/signals.py:58]: Nit / Future PR: a utility function to create this hash, with tests

```python
class Profile:
   ...
   def hashed_uid(self):
       return sha256(self.fxa.uid.encode("utf-8")).hexdigest()
```

File: emails/tracker_lists/level-one-tracker.json [hunk 3]
```diff
@@ -1 +1 @@
...
...omeriomail.com", "edrone.me", "emaillabs.co", "engagingnetworks.app", "ot.gliq.com", "track.gliq.com", "doubleclick.net", "google-analytics.com", "gasv1.com", "infernotions.com", "email-messaging.com", "ixactcontact.com", "mailer.lassocrm.com", "leadersend.com", "track-mb.bra2hmail.com", "liadm.com", "m3651.net", "m3652.net", "mailblue.eu", "eds5.mailcamp.nl", "agentofficemail.com", "mandrillapp.com", "mailcamp.net.pl", "mailtracker.pl", "senderit.pl", "xylionmail.pl", "mathtag.com", "api-01.moengage.com", "agkn.com", "myvisualiq.net", "nyl.as", "openeducat.org", "pvd.to", "pstmrk.it", "pushnami.com", "opens.responder.co.il", "system.send24.pl", "tracking.retailrocket.net", "sailplay.ru", "krxd.net", "pixel.inbox.exacttarget.com", "ondemand.com", "segment.io", "sendcloud.net", "sendfox.com", "sendiio.app", "sendpul.se", "sitescout.com", "smlists.com", "smartsendy.com", "smtp2go.com", "smtp2go.net", "servedbyadbutler.com", "strikestack.com", "hodes.com", "tealiumiq.com", "adventure-novels.com", "returnpath.net", "vinc.fr", "viralhosts.com", "trac.visme.co", "stable.cz", "campaign-tracking.woowup.com", "e.wordfly.com", "xpressmail.hu", "xtremepush.com", "em.yotpo.com", "p.yotpo.com"]
```
Comment (jwhitlock) [inline emails/tracker_lists/level-one-tracker.json:1]: Can you run this file through a JSON pretty-printer like `jq`?

File: emails/tracker_lists/level-two-tracker.json [hunk 4]
```diff
@@ -1 +1 @@
...
...m", "autoklose.com", "bitrix24.com", "exponea.com", "getblueshift.com", "boomeranggmail.com", "bounceexchange.com", "campaignmonitor.com", "carrotquest.io", "cirrusinsight.com", "connectif.cloud", "rs6.net", "customeriomail.com", "didtheyreadit.com", "buttondown.email", "emailtracker.website", "engagebay.com", "ezymarketer.net", "emailinc.net", "followup.cc", "getnotify.com", "gliq.com", "gmelius.com", "hunter.io", "lahar.com.br", "lassocrm.com", "leadboxer.com", "bra2hmail.com", "mailcamp.nl", "list-manage.com", "tinyletter.com", "mailerjet.com", "mailerlite.com", "mailspice.com", "mailtag.io", "mailtrack.io", "mixmax.com", "moengage.com", "outreach.io", "persistiq.com", "polymail.io", "propelleremail.co.uk", "responder.co.il", "send24.pl", "reply.io", "retailrocket.net", "revenuegrid.com", "smartcloudconnect.io", "rightinbox.com", "sailthru.com", "exacttarget.com", "pardot.com", "saleshandy.com", "salesloft.com", "streak.com", "substack.com", "tradiewebguys.com.au", "sendgrid.net", "user.com", "visme.co", "vocus.io", "woowup.com", "wordfly.com", "yesware.com", "yotpo.com", "zendable.com", "maillist-manage.com", "maillist-manage.com.au", "maillist-manage.eu", "maillist-manage.in"]
```
Comment (jwhitlock) [inline emails/tracker_lists/level-two-tracker.json:1]: Can you run this file through a JSON pretty-printer like `jq`?

**Review by jwhitlock — CHANGES_REQUESTED** [review]
Thanks @say-yawn! It is a good idea to track feature usage, especially on launch. I'm not sure if we want to do this with a signal, but I can't think of a better way without a couple days experimenting.

Can you add some tests?
* Saving a `Profile` enabling the feature - emits metric and log
* Saving a `Profile` disabling the feature - emits metric and log
* Saving a `Profile` without changing the feature (such as clearing `server_storage`) - no metric or log

File: emails/signals.py (hunk 1 above)
Comment (say-yawn) [reply emails/signals.py:49]: OOPS. Thanks for catching that.

File: emails/signals.py (hunk 2 above)
Comment (say-yawn) [reply emails/signals.py:58]: Moving this as a TODO. Do you want me to create an issue?

File: emails/tracker_lists/level-two-tracker.json (hunk 4 above)
Comment (say-yawn) [reply emails/tracker_lists/level-two-tracker.json]: [cbdab0c](https://github.com/mozilla/fx-private-relay/pull/2268/commits/cbdab0c7c897cf9f2b12e70f8d680e2beffb3c25)

File: emails/tracker_lists/level-one-tracker.json (hunk 3 above)
Comment (say-yawn) [reply emails/tracker_lists/level-one-tracker.json]: cbdab0c

File: emails/signals.py (hunk 2 above)
Comment (say-yawn) [reply emails/signals.py:58]: 383fc57

File: emails/signals.py (hunk 1 above)
Comment (say-yawn) [reply emails/signals.py]: 383fc57

**Review by jwhitlock — APPROVED** [review]
Thanks @say-yawn! I ran the server and confirmed that the log is emitted on disable and enable:

```
{"Timestamp": 1659124832559140096, "Type": "eventsinfo", "Logger": "fx-private-relay", "Hostname": "johnwhitlock-4zq05n", "EnvVersion": "2.0", "Severity": 6, "Pid": 95415, "Fields": {"enabled": false, "hashed_uid": "2105051f0ac74ee3caf3db7fb3f9030ed27223c5a530f1340a2fd16b414f2c7d", "msg": "tracker_removal_feature"}}
...
{"Timestamp": 1659124869618438912, "Type": "eventsinfo", "Logger": "fx-private-relay", "Hostname": "johnwhitlock-4zq05n", "EnvVersion": "2.0", "Severity": 6, "Pid": 95415, "Fields": {"enabled": true, "hashed_uid": "2105051f0ac74ee3caf3db7fb3f9030ed27223c5a530f1340a2fd16b414f2c7d", "msg": "tracker_removal_feature"}}
```

**Comment by say-yawn** [comment]
This change was merged and released as part of tag [2022.7.26.2](https://github.com/mozilla/fx-private-relay/releases/tag/2022.07.26.2).

**Pull request merged**